package/MDAnalysis/core/Selection.py

```diff
@@ -57,7 +58,106 @@ def unique(ag):
         return ag
 
 
+def is_keyword(val):
+    """Is val a selection keyword?
+
+    Returns False on any of the following strings:
+      - keys in SELECTIONDICT (tokens from Selection objects)
+      - keys in OPERATIONS (tokens from LogicOperations)
+      - (Parentheses)
+      - The value `None` (used as EOF in selection strings)
+    """
+    return (val in _SELECTIONDICT or
+            val in _OPERATIONS or
+            val in ['(', ')'] or
+            val is None)
+
+
+def grab_not_keywords(tokens):
+    """Pop tokens from the left until you hit a keyword
+
+    Parameters
+    ----------
+    tokens : collections.deque
+        deque of strings, some tokens some not
+
+    Returns
+    -------
+    values : list of strings
+        All non keywords found until a keyword was hit
+
+    Note
+    ----
+    This function pops the values from the deque
+
+    Examples
+    --------
+    grab_not_keywords(['H', 'and','resname', 'MET'])
+    >>> ['H']
+
+    grab_not_keywords(['H', 'Ca', 'N', 'and','resname', 'MET'])
+    >>> ['H', 'Ca' ,'N']
+
+    grab_not_keywords(['and','resname', 'MET'])
+    >>> []
+    """
+    values = []
+    while not is_keyword(tokens[0]):
+        val = tokens.popleft()
+        # Insert escape characters here to use keywords as names?
```

Comment:
It's just a placeholder so we know where we *could* put this.

So I think there's 2 situations,
 * if pdb reads whitespace (see #647) then we need to be able to include whitespace in the name
 * if someone had the brilliant idea of naming their atoms after a keyword :D